src/sync/installation.ts

```diff
@@ -95,6 +95,8 @@ export const updateTaskStatusAndContinue = async (
 
 	const updateRepoSyncFields: { [x: string]: string | Date} = { [getStatusKey(task.task)]: status };
 
+	const duration = task.startTime ? Date.now() - task.startTime : NaN;
```

Comment:
Why are we defaulting to `NaN` here? The second param needs to be a number `histogram(stat: string | string[], value: number, tags?: Tags, callback?: StatsCb): void;`

Reply:
haha, too late.

Reply:
Can you update in a new PR? I don't think we should be potentially passing a type that isn't supported.

Reply:
> Why are we defaulting to `NaN` here? The second param needs to be a number `histogram(stat: string | string[], value: number, tags?: Tags, callback?: StatsCb): void;`

Sorry, let me explain, I can push more code if need to.
So that typescript required it to be non undefined, has to be a number. And in js, `NaN` is considered to be a number.

My only concern is whether that `NaN` will impact the metrics. I hope not.
Otherwise, I have to say:
```
if (duration) 
  statsd.histogram(xxx, duration, xxx)
else
  statsd.histogram(xxx, xxx)
```
which make the code a bit messy.

Reply:
I'd prefer this tbh rather than having javascript witchery